service/discovery/openstack/compute_handle.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Fraunhofer AISEC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//           $$\                           $$\ $$\   $$\
+//           $$ |                          $$ |\__|  $$ |
+//  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+// $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+// $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+// $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+// \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+//  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+//
+// This file is part of Clouditor Community Edition.
+
+package openstack
+
+import (
+	"fmt"
+
+	"clouditor.io/clouditor/v2/api/discovery"
+	"clouditor.io/clouditor/v2/api/ontology"
+	"clouditor.io/clouditor/v2/internal/util"
+
+	"github.com/gophercloud/gophercloud/v2/openstack/compute/v2/servers"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// handleServer creates a virtual machine resource based on the Clouditor Ontology
+func (d *openstackDiscovery) handleServer(server *servers.Server) (ontology.IsResource, error) {
+	var (
+		err         error
+		bootLogging *ontology.BootLogging
+		osLogging   *ontology.OSLogging
+	)
+
+	// TODO(anatheka): Check again!
+	// // boot and os logging are logged together in the console log
+	// consoleOutput := servers.ShowConsoleOutput(context.Background(), d.clients.computeClient, server.ID, servers.ShowConsoleOutputOpts{})
+	// if consoleOutput.Result.StatusCode PrettyPrintJSON() != "" {
+	// 	bootLogging = &ontology.BootLogging{
+	// 		Enabled: true,
+	// 	}
+	// 	osLogging = &ontology.OSLogging{
+	// 		Enabled: true,
+	// 	}
+	// }
+
+	r := &ontology.VirtualMachine{
+		Id:           server.ID,
+		Name:         server.Name,
+		CreationTime: timestamppb.New(server.Created),
+		GeoLocation: &ontology.GeoLocation{
+			Region: "unknown", // TODO: Can we get the region?
```

Comment:
I can't beliieve it but I also do not find any information to get the region for a VM/Server. Maybe we have to dig a little bit deeper